api/controllers/game/ready.js

```diff
@@ -1,55 +1,54 @@
-const userAPI = sails.hooks['customuserhook'];
+module.exports = async function (req, res) {
+  try {
+    // Query for game and users
+    const game =  await sails.helpers.lockGame(req.session.game);
+    const players = await User.find({game: req.session.game}).sort('pNum');
+    const user = players[ req.session.pNum ];
+    game.players = players;
 
-module.exports = function (req, res) {
-  if (req.session.game && req.session.usr) {
-    const promiseGame = gameAPI.findGame(req.session.game);
-    const promiseUser = userAPI.findUser(req.session.usr);
-    Promise.all([promiseGame, promiseUser])
-      // Assign player readiness
-      .then(function foundRecords(values) {
-        const [game, user] = values;
-        let { pNum } = user;
-        let bothReady = false;
-        const gameUpdates = {};
-        switch (pNum) {
-          case 0:
-            gameUpdates.p0Ready = !game.p0Ready;
-            if (game.p1Ready) {
-              bothReady = true;
-            }
-            break;
-          case 1:
-            gameUpdates.p1Ready = !game.p1Ready;
-            if (game.p0Ready) {
-              bothReady = true;
-            }
-            break;
+    // Determine who is ready
+    let { pNum } = user;
+    let bothReady = false;
+    const gameUpdates = {};
+    switch (pNum) {
+      case 0:
+        gameUpdates.p0Ready = !game.p0Ready;
+        if (game.p1Ready) {
+          bothReady = true;
         }
-        if (bothReady) {
-          // Inform all clients this game has started
-          sails.sockets.blast('gameStarted', { gameId: game.id });
-          // Ensure game is no longer available for new players to join
-          gameUpdates.status = gameService.GameStatus.STARTED;
-          // Create Cards
-          return gameService.dealCards(game, gameUpdates);
+        break;
+      case 1:
+        gameUpdates.p1Ready = !game.p1Ready;
+        if (game.p0Ready) {
+          bothReady = true;
         }
-        Game.publish([game.id], {
-          change: 'ready',
-          userId: user.id,
-          pNum: user.pNum,
-          gameId: game.id,
-        });
-        return Game.updateOne({ id: game.id }).set(gameUpdates);
-      }) //End foundRecords
-      .then(function respond() {
-        return res.ok();
-      })
-      .catch(function failed(err) {
-        return res.badRequest(err);
+        break;
+    }
+
+    // Start game if both players are ready
+    if (bothReady) {
+      // Inform all clients this game has started
+      sails.sockets.blast('gameStarted', { gameId: game.id });
+      // Ensure game is no longer available for new players to join
+      gameUpdates.status = gameService.GameStatus.STARTED;
+      // Deal cards
+      await gameService.dealCards(game, gameUpdates);
+    } else {
+      // Send socket message
+      Game.publish([game.id], {
+        change: 'ready',
+        userId: user.id,
+        pNum: user.pNum,
+        gameId: game.id,
```

Comment:
I added this else block because sending the socket message for 'ready' after we initialize the deck and deal the cards was causing a bug where the client would update to say that one of the players wasn't ready and therefore triggering an overlay that said "Waiting for game to start" even when the game was clearly initialized.

A bit of digging revealed that generally our setup for when to show an overlay that the game isn't ready is pretty spotty and disorganized. We have one such overlay in the `GameOverlays`, one in `RematchView`, and one in the `GameUnavailableView`, all of which have their own conditional logic. We should write up an issue to reorganize those so they are easier to troubleshoot and reason about.

For now though this prevents the bug

Reply:
https://github.com/cuttle-cards/cuttle/issues/806